charts/prometheus-pushgateway/templates/statefulset.yaml

```diff
@@ -0,0 +1,121 @@
+{{- if .Values.runAsStatefulSet }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "prometheus-pushgateway.fullname" . }}
+  labels:
+{{ template "prometheus-pushgateway.defaultLabels" merge (dict "extraLabels" .Values.podLabels) .  }}
+spec:
```

Comment:
imho this could be a template as the spec is pretty much the same as for the deployment so both coudl use a template file here, to deduplicate code.

Reply:
I published a new commit using helpers file for this purpose. I'm kind of unsure of the change because I find it harder to maintain (for the additional complexity, I mean), but I get what you mean. I'm open to changes in how I structured tough

Reply:
I thought of the podspec. 
Added a commit, to show what i mean.
Maybe the volumes part can be imprioved to be in the podspec too.

Reply:
Thanks for your help. I agree with your changes. Regarding the volumes, I pushed a new commit for putting them inside the podspec